### PR TITLE
trivial: Rename meson option tpm->plugin_tpm

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/synaptics-rmi/meson.build [hunk 1]
```diff
@@ -1,3 +1,4 @@
+if get_option('gudev') and get_option('plugin_synaptics_rmi')
```
Comment (superm1) [inline plugins/synaptics-rmi/meson.build:1]: this should follow the pattern of showing an error if plugin flag set but not gudev set

File: plugins/meson.build [hunk 2]
```diff
@@ -1,149 +1,66 @@
...
 subdir('ep963x')
+subdir('fastboot')
+subdir('flashrom')
+subdir('fresco-pd')
+subdir('goodix-moc')
+subdir('hailuck')
 subdir('iommu')
+subdir('jabra')
 subdir('linux-lockdown')
 subdir('linux-sleep')
```
Comment (superm1) [inline plugins/meson.build:29]: I still think that we should make it clear that a few plugins like `linux-lockdown` and `linux-sleep` are truly linux only like my original PR had.

**Review by superm1 — COMMENTED** [review]
overall I think this is a better approach, yes

File: plugins/meson.build (hunk 2 above)
Comment (hughsie) [reply plugins/meson.build:29]: Ohh, I missed that bit. Got a list offhand of the ones you worked out were linux specific?

Comment (superm1) [reply plugins/meson.build:29]: These were the ones I made out, feel free to disagree or add anything else you think though.

```
  subdir('acpi-dmar')       #/sys/firmware/acpi
  subdir('acpi-facp')       #/sys/firmware/acpi
  subdir('iommu')           #linux kernel iommu
  subdir('linux-lockdown')
  subdir('linux-sleep')
  subdir('linux-swap')
  subdir('linux-tainted')
  subdir('pci-bcr')         #pci
  subdir('pci-mei')         #pci
  subdir('upower')
  subdir('pixart-rf')       #linux kernel hidraw
```

Comment (hughsie) [reply plugins/meson.build:29]: Do we want to run acpi-facp-self-test on !linux? I'm erring on yes.

File: plugins/synaptics-rmi/meson.build (hunk 1 above)
Comment (hughsie) [reply plugins/synaptics-rmi/meson.build:1]: Agree, fixed.

**Review by superm1 — APPROVED** [review]
Lgtm now

**Pull request merged**